group-coordinator/src/test/java/org/apache/kafka/coordinator/group/taskassignor/StickyTaskAssignorTest.java

```diff
@@ -0,0 +1,1311 @@
+
+        final Map<String, AssignmentMemberSpec> members = mkMap(
+                mkEntry("member1", memberSpec1), mkEntry("member2", memberSpec2), mkEntry("newMember", newMemberSpec));
+        GroupAssignment result = assignor.assign(
+                new GroupSpecImpl(members, Collections.singletonList("test-subtopology"), new HashMap<>()),
+                x -> 6
+        );
+
+        final List<Integer> mem1Tasks = getAllActiveTaskIds(result, "member1");
+        assertFalse(mem1Tasks.contains(3));
+        assertFalse(mem1Tasks.contains(4));
+        assertFalse(mem1Tasks.contains(5));
+        assertEquals(2, mem1Tasks.size());
+
+        final List<Integer> mem2Tasks = getAllActiveTaskIds(result, "member2");
+        assertFalse(mem2Tasks.contains(0));
+        assertFalse(mem2Tasks.contains(1));
+        assertFalse(mem2Tasks.contains(2));
+        assertEquals(2, mem2Tasks.size());
+
+        assertEquals(2, getAllActiveTaskIds(result, "newMember").size());
+    }
+
+    @Test
+    public void shouldAssignTasksToNewClientWithoutFlippingAssignmentBetweenExistingAndBouncedClients() {
+        final AssignmentMemberSpec memberSpec1 = createAssignmentMemberSpec("process1", mkMap(mkEntry("test-subtopology", mkSet(0, 1, 2, 6))), Collections.emptyMap());
+        final AssignmentMemberSpec memberSpec2 = createAssignmentMemberSpec("process2", Collections.emptyMap(), mkMap(mkEntry("test-subtopology", mkSet(3, 4, 5))));
+        final AssignmentMemberSpec newMemberSpec = createAssignmentMemberSpec("newProcess");
+
+        final Map<String, AssignmentMemberSpec> members = mkMap(
+                mkEntry("member1", memberSpec1), mkEntry("member2", memberSpec2), mkEntry("newMember", newMemberSpec));
+        GroupAssignment result = assignor.assign(
+                new GroupSpecImpl(members, Collections.singletonList("test-subtopology"), new HashMap<>()),
+                x -> 7
+        );
+
+        final List<Integer> mem1Tasks = getAllActiveTaskIds(result, "member1");
+        assertFalse(mem1Tasks.contains(3));
+        assertFalse(mem1Tasks.contains(4));
+        assertFalse(mem1Tasks.contains(5));
+        assertEquals(3, mem1Tasks.size());
+
+        final List<Integer> mem2Tasks = getAllActiveTaskIds(result, "member2");
+        assertFalse(mem2Tasks.contains(0));
+        assertFalse(mem2Tasks.contains(1));
+        assertFalse(mem2Tasks.contains(2));
+        assertEquals(2, mem2Tasks.size());
+
+        assertEquals(2, getAllActiveTaskIds(result, "newMember").size());
+    }
+
+    @Test//todo rack aware
+    public void shouldViolateBalanceToPreserveActiveTaskStickiness() {
+        final AssignmentMemberSpec memberSpec1 = createAssignmentMemberSpec("process1", mkMap(mkEntry("test-subtopology", mkSet(0, 1, 2))), Collections.emptyMap());
+        final AssignmentMemberSpec memberSpec2 = createAssignmentMemberSpec("process2");
+    }
+
+    @Test//todo rack aware
```

Comment:
What do you mean by this? We will not implement rack aware assignment for now, so we can probably remove those?